gateway/handle.go

```diff
@@ -614,3 +614,228 @@ func (gw *Handle) addToWebRequestQ(_ *http.ResponseWriter, req *http.Request, do
 	}
 	userWebRequestWorker.webRequestQ <- &webReq
 }
+
+func (gw *Handle) writeToJobsDB() http.HandlerFunc {
```

Comment:
I believe this function is doing too many things. I would break the payload preparation logic from http handler.

HTTP handler:
* expose `func(w http.ResponseWriter, r *http.Request)`
* validates payload
* call method to `writeToJobsDB`
* HTTP response, and status code

writeToJobsDB:
* extract params
* handle suppression
* rsourcesStats
* write to jobsDB

We can decouple `writeToJobsDB` from `gateway.Handle` struct, it will depend on suppression service and jobsDB. It should not be aware of HTTP.